tests/test_engine.py

```diff
@@ -1274,6 +1277,27 @@ def my_flow():
 
         assert i <= 10, "`just_sleep` should not be running after timeout"
 
+    async def test_report_flow_run_crashes_handles_sigterm(
+        self, flow_run, orion_client, monkeypatch
+    ):
+        original_handler = lambda args: args
+        signal_receiver = Mock(return_value=original_handler)
+        monkeypatch.setattr("signal.signal", signal_receiver)
+
+        async with report_flow_run_crashes(flow_run=flow_run, client=orion_client):
+            assert signal_receiver.call_args_list[0][0][0] == signal.SIGTERM
+            signal_handler = signal_receiver.call_args_list[0][0][1]
+
+            # Call the signal handler and expect that it raises a
+            # `TerminationSignal` exception.
+            with pytest.raises(TerminationSignal):
+                signal_handler()
+
```

Comment:
I think this test should be updated to register a new signal handler then assert that it is called when the context exits to ensure that the "existing" signal handler is called.